### PR TITLE
Adding admin view of feedback.

### PR DESCRIPTION
**Pull request by jmacdotorg**
New action: /admin/feedback

Displays all collected feedback, ordered by judge.

Locked to the 'curator' role because I didn't feel like introducing a new role for this? May change mind on that later.

### CODE REVIEW

File: IFComp/lib/IFComp/Controller/Admin/Feedback.pm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+use Moose;
+use namespace::autoclean;
+
+BEGIN { extends 'Catalyst::Controller'; }
+
+sub index : Chained('/admin/root') : PathPart( 'feedback') : Args(0) {
+    my ( $self, $c ) = @_;
+
+    unless ( $c->user && $c->check_any_user_role('curator') ) {
+        $c->log->debug("get outta here");
```
Comment (adherzog) [inline IFComp/lib/IFComp/Controller/Admin/Feedback.pm]: Please get rid of this line, or improve the log message and add `if ($c->debug)` to the line.

File: IFComp/root/src/admin/feedback/index.tt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Here is all the feedback received this year, sorted by judge name.</p>
+
+<p><strong>Please consider the contents of this page sensitive information.</strong> Outside of this page, only the author of a given entry can see the feedback left for it, and in that case they cannot see the identity of the judge who wrote it (unless the judge volunteered this information in the text of the feedback itself.)
+</p>
+
+<hr />
+
+[% IF feedback_rs.count %]
+    [% WHILE (feedback = feedback_rs.next) %]
+        <p><strong>[% feedback.judge.name %] ([% feedback.judge.email %])</strong>, on [% feedback.entry.title %]:</p>
```
Comment (adherzog) [inline IFComp/root/src/admin/feedback/index.tt]: Why aren't the name, email, and title being passed through the `html` filter?

File: IFComp/root/src/admin/index.tt [hunk 3]
```diff
@@ -10,4 +10,8 @@
         <li><a href="[% c.uri_for('/admin/voting/') %]">Voting</a></li>
      </ul>
      
+     <ul>
+        <li><a href="[% c.uri_for('/admin/feedback/') %]">Feedback</a></li>
```
Comment (adherzog) [inline IFComp/root/src/admin/index.tt]: If looks like the admin menu is now available to both votecounters and curators, but they can each access only one of the two available menu options. Let's show/hide the links based on whether the user has the appropriate role.

File: IFComp/root/src/admin/feedback/index.tt (hunk 2 above)
Comment (jmacdotorg) [reply IFComp/root/src/admin/feedback/index.tt]: I feel comfortable saying that if these fields are allowed to contain HTML-unsafe characters then we have bigger problems. And maybe we do! But if so they should be addressed as their own issue, and then tackled at the form-validation level.

Comment (adherzog) [reply IFComp/root/src/admin/feedback/index.tt]: You wouldn't allow an ampersand in an entry title?

The title and author are already passed through the `html` filter on the ballot page. There's no reason not to do it here, too, even if you don't expect it to be required. (In fact, why not re-use the existing `_entry_title.tt` and `author_name.tt` templates?)

Comment (jmacdotorg) [reply IFComp/root/src/admin/feedback/index.tt]: You're right, was confusing the HTML filter for tag-stripping.

**Pull request merged**